src/context.rs

```diff
@@ -275,9 +275,8 @@ impl<A: Actor> Context<A> {
                 let mut next_msg = self.next_message();
                 match future::select(fut, &mut next_msg).await {
                     Either::Left((future_res, _)) => {
-                        if let Some(msg) = next_msg.cancel() {
-                            self.tick(msg, actor).await;
-                        }
+                        let message = next_msg.await; // If don't await the future, message ordering might be compromised.
```

Comment:
Consider the following:

- Future is `futures::future::ok(())`
- There is one address which is sleeping for 10 minutes and will send a message later

In this case, this `select` would wait the full 10 minutes

Reply:
Right, in this case, `cancel` would have returned `None` right?

Reply:
Yea. On the other hand, I think `now_or_never` could race, on the other hand...

Reply:
I see the problem, so this can only be solved if:

- We accept the potential re-ordering of messages
- We somehow make sure we can re-insert messages into the mailbox without re-ordering

Reply:
> Yea. On the other hand, I think `now_or_never` could race, on the other hand...

Oh, that is an interesting idea! Whether or not this works might depend on how many `poll`s we need to dispatch? Do you have any idea on how we could test this?

Reply:
I managed to write a test! Let me know what you think.